software/controller/lib/hal/gpio_mock.cpp

```diff
@@ -21,13 +21,11 @@ Reference abbreviations [RM], [DS], etc are defined in hal/README.md.
 */
 #if !defined(BARE_STM32)
```

Comment:
Yes, but the problem is that we are still using ifdefs instead of implementing an API defined in abstract class. This will require further work to remove this. I have added a TODO into gpio_stm32.cpp about this and leaving this next step for another PR